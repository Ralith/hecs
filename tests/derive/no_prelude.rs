#![no_implicit_prelude]

#[derive(::hecs::Bundle)]
struct Foo {
    foo: (),
}

fn main() {}
