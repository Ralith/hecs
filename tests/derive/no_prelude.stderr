error[E0433]: failed to resolve: use of undeclared crate or module `std`
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ use of undeclared crate or module `std`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0432]: unresolved import `std`
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ use of undeclared crate or module `std`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find macro `panic` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find macro `vec` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find trait `FnOnce` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::ops::FnOnce;
  |
4 | use std::ops::FnOnce;
  |

error[E0412]: cannot find type `Vec` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this struct
  |
4 | use std::vec::Vec;
  |

error[E0405]: cannot find trait `FnMut` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::ops::FnMut;
  |
4 | use std::ops::FnMut;
  |

error[E0412]: cannot find type `TypeId` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::any::TypeId;
  |
4 | use std::any::TypeId;
  |

error[E0433]: failed to resolve: use of undeclared crate or module `std`
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in `std::collections`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use hashbrown::HashSet;
  |
4 | use std::collections::HashSet;
  |

error[E0433]: failed to resolve: use of undeclared crate or module `std`
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in `std::any`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::any::TypeId;
  |
4 | use std::any::TypeId;
  |

error[E0433]: failed to resolve: use of undeclared type `TypeId`
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::any::TypeId;
  |
4 | use std::any::TypeId;
  |

error[E0412]: cannot find type `Option` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::option::Option;
  |
4 | use std::option::Option;
  |

error[E0412]: cannot find type `Result` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::fmt::Result;
  |
4 | use core::result::Result;
  |
4 | use std::fmt::Result;
  |
4 | use std::io::Result;
  |
    and 2 other candidates

error[E0425]: cannot find function, tuple struct or tuple variant `Ok` in this scope
 --> $DIR/no_prelude.rs:3:10
  |
3 | #[derive(::hecs::Bundle)]
  |          ^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
4 | use core::result::Result::Ok;
  |
4 | use std::result::Result::Ok;
  |
